### PR TITLE
feat(python): 30x speedup initialising `Series` from python `range` object

### PR DESCRIPTION
**Pull request by alexander-beedie**
Took a quick look at #5395 (on my lunch break, hah :) and spotted an optimisation opportunity; can automatically substitute python `range` for polars' `arange` expression.

```python
import polars as pl
s = pl.Series( range(1_000_000_000) )
```
**Before:** `31.87s`
**After:** `1.045s`

FYI: looks like pandas was already doing a similar optimisation; we were much slower, and now we're ~40% faster (pandas timing for Series init with the same range object: `1.47s`).

Can look at the actual issue itself later this evening.

### CODE REVIEW

File: py-polars/polars/internals/lazy_functions.py [hunk 1]
```diff
@@ -1387,11 +1387,13 @@ def arange(
     """
     low = pli.expr_to_lit_or_expr(low, str_to_lit=False)
     high = pli.expr_to_lit_or_expr(high, str_to_lit=False)
-
     if eager:
-        df = pli.DataFrame({"a": [1]})
-        return df.select(arange(low, high, step).alias("arange"))["arange"]
-
+        return (
```
Comment (ritchie46) [inline py-polars/polars/internals/lazy_functions.py:1391]: We could write 

```python
pl.arange(low, high, step, eager=True)
```

File: py-polars/polars/internals/lazy_functions.py [hunk 2]
```diff
@@ -1387,11 +1387,13 @@ def arange(
...
+            pli.DataFrame()
```
Comment (ritchie46) [inline py-polars/polars/internals/lazy_functions.py:1392]: idem

File: py-polars/polars/internals/series/series.py [hunk 3]
```diff
@@ -223,6 +223,13 @@ def __init__(
 
             if dtype is not None:
                 self._s = self.cast(dtype, strict=True)._s
+        elif isinstance(values, range):
+            self._s = (
```
Comment (ritchie46) [inline py-polars/polars/internals/series/series.py:227]: idem

**Review by ritchie46 — COMMENTED** [review]
So simple. :)

File: py-polars/polars/internals/lazy_functions.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/polars/internals/lazy_functions.py:1391]: I hang my head in shame for not spotting that, lol... will update accordingly.

File: py-polars/polars/internals/series/series.py (hunk 3 above)
Comment (alexander-beedie) [reply py-polars/polars/internals/series/series.py:230]: Updated (`eager=True`)

File: py-polars/polars/internals/lazy_functions.py (hunk 2 above)
Comment (alexander-beedie) [reply py-polars/polars/internals/lazy_functions.py:1392]: Setting `eager=True` here would be recursive.

**Pull request merged**